Makefile cannot build pre-existing codebases

If you checkout an existing Drupal site into `codebase` and do `make`, the build will fail with

```
Composer could not find a composer.json file in /app
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
make: *** [Makefile:15: drupal_init] Error 1
```

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -6,7 +6,7 @@ codebase="drupal"
 config_dir="$PWD/config/drupal"
 current_folder="$PWD"
 composer_general_flags="--ignore-platform-reqs --no-interaction"
-OS=`uname -s`
+OS=$(uname -s)
 [[ "$OS" == "Darwin" ]] && is_darwin=true || is_darwin=false
 
 function fail {
@@ -44,7 +44,7 @@ function download_drupal() {
 
   # Delete codebase just in case a previous command failed to finish the installation.
   [[ -d ./codebase ]] && rm -rf ./codebase
-  composer_cmd $args $composer_general_flags codebase
+  composer_cmd "$args" "$composer_general_flags" codebase
 
   download_required_packages
 }
@@ -67,7 +67,7 @@ function download_required_packages() {
     local pattern="    \"extra\""
     if $is_darwin; then
       local snippet=$'\    "autoload": {"files": ["load.environment.php"]},\n'
-      sed -i '' -e '\|^'"$pattern"'|i\'$'\n'"$snippet" composer.json
+      sed -i '' -e '\|^'"$pattern"'|i'\'''$'\n'"$snippet" composer.json
     else
       local snippet="\    \"autoload\": {\n       \"files\": [\"load\.environment\.php\"]\n    },\n"
       sed -i -e "/^${pattern}/i ${snippet}" composer.json
@@ -78,10 +78,10 @@ function download_required_packages() {
     local package=${packages[$i]}
     local version=${versions[$i]}
     # Only installing a package when it is not available in composer.json
-    if [[ ! $(grep "${package}" composer.json) ]]; then
+    if ! grep -q "${package}" composer.json; then
       echo "       Requiring ${package}. Downloading."
       echo " "
-      composer_cmd require ${package}:${version} $composer_general_flags
+      composer_cmd require "${package}":"${version}" "$composer_general_flags"
     else
       echo "       ${package} was found in the composer.json. Skipping."
     fi
@@ -120,25 +120,24 @@ function composer_cmd() {
   ###
   # Determine how we will be running composer.
   ###
-  local composer=$(command -v composer || true)
-  if [[ ! $composer ]]; then
+  if [[ ! $(command -v composer || true) ]]; then
     # We use the docker composer image to run composer related commands.
     echo >&2
     echo -e "\033[1m[INFO]\033[0m Using the official composer docker image to run composer commands"
     echo " "
     echo >&2
-    mkdir -p "$HOME/.composer"
+    mkdir -p "$HOME"/.composer
     if $is_darwin; then
-      docker container run -it --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3 $@
+      docker container run -it --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 "$@"
     else
-      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3 $@
+      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 "$@"
     fi
   else
     echo >&2
     echo -e "\033[1m[INFO]\033[0m Using local composer to run the commands"
     echo " "
     echo >&2
-    composer $@
+    composer "$@"
   fi
 }
 
@@ -167,7 +166,7 @@ done
 # Checking if the project code exists.
 ###
 if [[ ! -f "$current_folder/codebase/composer.json" ]]; then
-  download_drupal $codebase
+  download_drupal "$codebase"
 else
   download_required_packages
 fi
@@ -180,7 +179,7 @@ fi
 ###
 # Initialize drupal files persistent storage folders.
 ###
-mkdir -p $PWD/data/drupal/files/public $PWD/data/drupal/files/private
+mkdir -p "$PWD"/data/drupal/files/public "$PWD"/data/drupal/files/private
 
 ###
 # Running composer install just in case the user has an existing project.
@@ -192,7 +191,7 @@ if [[ ! -f "$current_folder/codebase/vendor/autoload.php" ]]; then
   echo " "
   echo >&2
 
-  composer_cmd install $composer_general_flags
+  composer_cmd install "$composer_general_flags"
   cd ..
 fi
 

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -27,9 +27,9 @@ function download_drupal() {
   local args="create-project born-digital/drupal-project:dev-isle8-dev"
   local codebase="$1"
 
-  if [[ ! $composer ]]; then
-    fail "We could not download drupal. Ensure composer or docker is setup and installed properly on your local host."
-  fi
+  # if [[ ! $composer ]]; then
+  #   fail "We could not download drupal. Ensure composer or docker is setup and installed properly on your local host."
+  # fi
 
   if [[ "$codebase" == "drupal" ]]; then
     local args="create-project drupal-composer/drupal-project:8.x-dev"
@@ -44,7 +44,7 @@ function download_drupal() {
 
   # Delete codebase just in case a previous command failed to finish the installation.
   [[ -d ./codebase ]] && rm -rf ./codebase
-  $composer $args $composer_general_flags codebase
+  composer_cmd $args $composer_general_flags codebase
 
   download_required_packages
 }
@@ -81,7 +81,7 @@ function download_required_packages() {
     if [[ ! $(grep "${package}" composer.json) ]]; then
       echo "       Requiring ${package}. Downloading."
       echo " "
-      $composer require ${package}:${version} $composer_general_flags
+      composer_cmd require ${package}:${version} $composer_general_flags
     else
       echo "       ${package} was found in the composer.json. Skipping."
     fi
@@ -116,6 +116,32 @@ function create_required_files() {
   fi
 }
 
+function composer_cmd() {
+  ###
+  # Determine how we will be running composer.
+  ###
+  local composer=$(command -v composer || true)
+  if [[ ! $composer ]]; then
+    # We use the docker composer image to run composer related commands.
+    echo >&2
+    echo -e "\033[1m[INFO]\033[0m Using the official composer docker image to run composer commands"
+    echo " "
+    echo >&2
+    mkdir -p "$HOME/.composer"
+    if $is_darwin; then
+      docker container run -it --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3 $@
+    else
+      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3 $@
+    fi
+  else
+    echo >&2
+    echo -e "\033[1m[INFO]\033[0m Using local composer to run the commands"
+    echo " "
+    echo >&2
+    composer $@
+  fi
+}
+
 while [ ! $# -eq 0 ]
 do
   case "$1" in
@@ -136,24 +162,6 @@ do
   esac
   shift
 done
-
-###
-# Determine how we will be running composer.
-###
-composer=$(command -v composer || true)
-if [[ ! $composer ]]; then
-  # We use the docker composer image to run composer related commands.
-  echo >&2
-  echo -e "\033[1m[INFO]\033[0m Using the official composer docker image to run composer commands"
-  echo " "
-  echo >&2
-  mkdir -p "$HOME/.composer"
-  if $is_darwin; then
-    composer="docker container run -it --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3"
-  else
-    composer="env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:$GUID -v $HOME/.composer:/tmp -v $PWD:/app composer:1.9.3"
-  fi
-fi
 
 ###
 # Checking if the project code exists.
@@ -179,7 +187,12 @@ mkdir -p $PWD/data/drupal/files/public $PWD/data/drupal/files/private
 ###
 if [[ ! -f "$current_folder/codebase/vendor/autoload.php" ]]; then
   cd "$current_folder/codebase"
-  $composer install $composer_flags
+  echo >&2
+  echo -e "\033[1m[INFO]\033[0m Running composer install just in case the user has an existing project"
+  echo " "
+  echo >&2
+
+  composer_cmd install $composer_general_flags
   cd ..
 fi
 

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-set -e
+set -euo pipefail
 
 codebase="drupal"
 config_dir="$PWD/config/drupal"

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -176,6 +176,10 @@ fi
 
 # load the config here
 if [[ "$codebase" == "islandora" && ! -f codebase/config/sync/core.extension.yml ]]; then
+  echo >&2
+  echo -e "\033[1m[INFO]\033[0m Extracting default drupal islandora configurations in codebase"
+  echo " "
+  echo >&2
   tar -xzf config/drupal/islandora-starter-config.tar.gz codebase/config/sync
 fi
 

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -129,10 +129,10 @@ function composer_cmd() {
     mkdir -p "$HOME"/.composer
     if $is_darwin; then
       # shellcheck disable=SC2068
-      docker container run -it --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
+      docker container run -it --rm --user $UID:"$(id -g)" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
     else
       # shellcheck disable=SC2068
-      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
+      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:"$(id -g)" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
     fi
   else
     echo >&2

--- a/scripts/drupal/init.sh
+++ b/scripts/drupal/init.sh
@@ -128,16 +128,19 @@ function composer_cmd() {
     echo >&2
     mkdir -p "$HOME"/.composer
     if $is_darwin; then
-      docker container run -it --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 "$@"
+      # shellcheck disable=SC2068
+      docker container run -it --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
     else
-      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 "$@"
+      # shellcheck disable=SC2068
+      env MSYS_NO_PATHCONV=1 docker container run -t --rm --user $UID:"$GUID" -v "$HOME"/.composer:/tmp -v "$PWD":/app composer:1.9.3 $@
     fi
   else
     echo >&2
     echo -e "\033[1m[INFO]\033[0m Using local composer to run the commands"
     echo " "
     echo >&2
-    composer "$@"
+    # shellcheck disable=SC2068
+    composer $@
   fi
 }
 

